### PR TITLE
Cleanup Makefile and general project setup

### PR DESCRIPTION
**Pull request by darinkrauss**
@jh-bate

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,5 @@
 language: go
 
-services:
-  - mongodb
```
Comment (jh-bate) [inline .travis.yml]: so pretty sure this is needed

**Comment by jh-bate** [comment]
one query other than that LGTM

File: .travis.yml (hunk 1 above)
Comment (darinkrauss) [inline .travis.yml]: It is, but my earlier integration added a second copy of it, so this deletes the extra copy. :)

Comment (jh-bate) [inline .travis.yml]: ahh - sweet then.

**Pull request closed without merging**